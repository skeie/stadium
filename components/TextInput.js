// @flow

import React from 'react';
import { TextInput } from 'react-native';
import { styles } from './StyledText';
import colors from '../constants/Colors';

type Props = any; // https://facebook.github.io/react-native/docs/textinput.html

const Input = (props: Props) => {
  return (
    <TextInput
      autoGrow
      underlineColorAndroid="transparent"
      placeholderTextColor={colors.placeholder}
      {...props}
      style={[styles.style, props.type && styles[props.type], props.style]}
    />
  );
};

export default Input;
