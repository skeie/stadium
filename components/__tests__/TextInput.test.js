import 'react-native';
import React from 'react';
import TextInput from '../TextInput';
import renderer from 'react-test-renderer';

it('TextInput renders correctly', () => {
    const tree = renderer.create(<TextInput />).toJSON();
    expect(tree).toMatchSnapshot();
});

