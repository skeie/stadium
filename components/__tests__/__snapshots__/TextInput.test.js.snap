// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  autoGrow={true}
  placeholderTextColor="#6B5F6B"
  style={
    Array [
      Object {
        "color": "#553555",
        "fontFamily": "poppins",
      },
      undefined,
      undefined,
    ]
  }
  underlineColorAndroid="transparent"
/>
`;
